internal/handlers/handler_oidc_userinfo.go

```diff
@@ -0,0 +1,79 @@
+package handlers
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/google/uuid"
+	"github.com/ory/fosite"
+	fositejwt "github.com/ory/fosite/token/jwt"
+
+	"github.com/authelia/authelia/internal/middlewares"
+	"github.com/authelia/authelia/internal/oidc"
+)
+
+func oidcUserinfo(ctx *middlewares.AutheliaCtx, rw http.ResponseWriter, req *http.Request) {
+	session := newOpenIDSession("")
+
+	tokenType, ar, err := ctx.Providers.OpenIDConnect.Fosite.IntrospectToken(req.Context(), fosite.AccessTokenFromRequest(req), fosite.AccessToken, session)
```

Comment:
Are you sure we are not mixing token introspection and userinfo endpoints?

Reply:
I am fairly certain. I believe in order to obtain userinfo you need to do introspection.